bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/discovery/EnOceanDeviceDiscoveryService.java

```diff
@@ -193,9 +190,12 @@ private Integer sendTeachInResponse(ERP1Message msg, String enoceanId) {
             // send response
             EEP response = EEPFactory.buildResponseEEPFromTeachInERP1(msg, newSenderId, true);
             if (response != null) {
-                bridgeHandler.sendMessage(response.getERP1Message(), null);
-                logger.debug("Teach in response for {} with new senderId {} (= offset {}) sent", enoceanId,
-                        HexUtils.bytesToHex(newSenderId), offset);
+                BasePacket bPacket = response.getERP1Message();
+                if (bPacket != null) {
+                    bridgeHandler.sendMessage(bPacket, null);
```

Comment:
Unfortunately the original code is using null values for signalling exceptions, which makes this new code more verbose. Did you thought about fixing this all at once? Otherwise, there will be another big PR refactoring the code to make use of exceptions.

Reply:
There is certainly room for improvement, but this is beyond the scope of this PR. I might look into the exception handling in the future, but for that i would need more Enocean devices to test them myself.

Reply:
I started to refactor another binding to add null annotations myself in the past ... and aborted. I realized that one needs to touch the whole code, if the binding wasn't written with null annotations in mind. Otherwise you end up with adding null checks where they aren't neccessary at all by design, but demanded by the compiler (injection, exceptions or interfaces declaring nullable, although [only one call has a nullable argument.](https://github.com/openhab/openhab-addons/pull/14023/files#diff-557e3fbed7d8ee77148027c1cac66f57de95ef5d3344490f2f6f86f734d0c180R275) Here: pushing down the channel configuration the entire call hierarchy and checking each occurrence in every implementing class for null). This bloats up the code, is prone to introduce regressions and is hard to review, just to fix the compiler warnings. Especially corner cases where something *can* be null are hard to test, though.

My approach is to leave code without null annotations as-is, unless you add features and do a big refactoring anyway. 

Just for clarity, I think null annotations are a good thing and don't bloat up the code, but only if the code is designed with null annotations in mind.

My suggestion for proceeding with this PR is to make further improvements in a follow-up PR, if you want to and only address the review feedback in this PR.

However, fixing the other checkstyle warnings (naming schemes) is a good thing!

Reply:
Thanks, i keep that in mind with other bindings that are on my list of adding null annotations. For this binding i'll look to get some devices and do more refactoring in another PR.